src/DynamicRupture/FrictionLaws/LinearSlipWeakening.h

```diff
@@ -157,30 +157,20 @@ class LinearSlipWeakeningLaw : public BaseFrictionLaw<LinearSlipWeakeningLaw<Spe
           -cohesion[ltsFace][pointIndex] -
           this->mu[ltsFace][pointIndex] * std::min(totalNormalStress, static_cast<real>(0.0));
 
-      specialization.strengthHook(strength[pointIndex],
-                                  this->slipRateMagnitude[ltsFace][pointIndex],
-                                  totalNormalStress,
-                                  this->mu[ltsFace][pointIndex],
-                                  this->deltaT[timeIndex],
-                                  ltsFace,
-                                  pointIndex);
+      strength[pointIndex] =
+          specialization.strengthHook(strength[pointIndex],
+                                      this->slipRateMagnitude[ltsFace][pointIndex],
+                                      this->deltaT[timeIndex],
+                                      ltsFace,
+                                      pointIndex);
     }
   }
 
   void calcStateVariableHook(std::array<real, misc::numPaddedPoints>& stateVariable,
                              unsigned int timeIndex,
                              unsigned int ltsFace) {
     alignas(ALIGNMENT) real resampledSlipRate[misc::numPaddedPoints]{};
-    dynamicRupture::kernel::resampleParameter resampleKrnl;
-    resampleKrnl.resample = init::resample::Values;
-    resampleKrnl.originalQ = this->slipRateMagnitude[ltsFace];
-    resampleKrnl.resampledQ = resampledSlipRate;
-
-    // Resample slip-rate, such that the state increment (slip) lies in the same polynomial space as
-    // the degrees of freedom resampleMatrix first projects LocSR on the two-dimensional basis on
-    // the reference triangle with degree less or equal than CONVERGENCE_ORDER-1, and then evaluates
-    // the polynomial at the quadrature points
-    resampleKrnl.execute();
+    specialization.resampleSlipRates(resampledSlipRate, this->slipRateMagnitude[ltsFace]);
```

Comment:
About this (L178):

```
// integrate slip rate to get slip = state variable
      this->accumulatedSlipMagnitude[ltsFace][pointIndex] +=
          resampledSlipRate[pointIndex] * this->deltaT[timeIndex];
```
Do you think resampledSlipRate could get negative because of the resampling?
Shouldn't we add a max(0, ) somewhere?